cmd/kubectl-gadget/utils/trace.go

```diff
@@ -597,7 +598,18 @@ func waitForCondition(traceID string, conditionFunction func(*gadgetv1alpha1.Tra
 	}
 
 	if err != nil {
-		return nil, err
+		if !errors.Is(err, wait.ErrWaitTimeout) {
+			return nil, err
+		}
+
+		// Print a message for traces that timed out
+		for _, trace := range traceList.Items {
+			_, satisfied := satisfiedTraces[trace.ObjectMeta.Name]
+			_, errored := erroredTraces[trace.ObjectMeta.Name]
+			if !satisfied && !errored {
+				fmt.Fprintf(os.Stderr, "Timeout waiting for condition on node %q\n", trace.Spec.Node)
```

Comment:
Beyond this change, I was thinking that `Timeout waiting for condition` (which is identical to the current error message), maybe is not that user-friendly. I think users will ask themselves, "_What condition?_". Maybe something like `Timeout trying to run gadget on node`? WDYT?

Reply:
The problem is that `waitForCondition` is used in `waitForTraceState` and `waitForNoOperation`, so `Timeout trying to run gadget on node` is misleading in the second case. I have no more ideas to fix it without a big change.

Reply:
Let's keep this for now.